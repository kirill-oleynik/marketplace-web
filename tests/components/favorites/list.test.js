const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const { FavoritesList } = require('../../../app/components/favorites/list');

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      <FavoritesList
        t={(translation) => translation}
        favorites={[
          {
            id: 1,
            application: {
              id: 2,
              title: 'Hello',
              summary: 'Hello World',
              slug: 'hello',
              logo: 'https://example.com'
            }
          }
        ]}
      />
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#handleExpandClick', () => {
  test('it toggles isExpanded state', () => {
    const component = shallow(
      <FavoritesList
        t={(translation) => translation}
        favorites={[
          {
            id: 1,
            application: {
              id: 2,
              title: 'Hello',
              summary: 'Hello World',
              slug: 'hello',
              logo: 'https://example.com'
            }
          }
        ]}
      />
    );

    expect(component.state().isExpanded).toEqual(false);
    component.instance().handleExpandClick();
    expect(component.state().isExpanded).toEqual(true);
    component.instance().handleExpandClick();
    expect(component.state().isExpanded).toEqual(false);
  });
});
