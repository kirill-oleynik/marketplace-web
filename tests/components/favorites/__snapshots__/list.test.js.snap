// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div>
  <Row
    tag="div"
  >
    <Col
      sm="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <ApplicationPreview
        application={
          Object {
            "id": 2,
            "logo": "https://example.com",
            "slug": "hello",
            "summary": "Hello World",
            "title": "Hello",
          }
        }
        remove={false}
        vertical={false}
      />
    </Col>
  </Row>
</div>
`;
